autogluon_zeroshot/simulation/repository.py

```diff
@@ -0,0 +1,183 @@
+from typing import Optional
+
+import pandas as pd
+import numpy as np
+
+from autogluon_zeroshot.contexts import get_context
+
+from autogluon_zeroshot.contexts.context_artificial import load_context_artificial
+from autogluon_zeroshot.loaders import Paths
+from typing import List
+from autogluon_zeroshot.simulation.ensemble_selection_config_scorer import EnsembleSelectionConfigScorer
+from autogluon_zeroshot.simulation.simulation_context import ZeroshotSimulatorContext
+from autogluon_zeroshot.simulation.tabular_predictions import TabularModelPredictions
+from autogluon_zeroshot.utils import catchtime
+
+
+class EvaluationRepository:
+    def __init__(
+            self,
+            zeroshot_context: ZeroshotSimulatorContext,
+            tabular_predictions: TabularModelPredictions,
+            ground_truth: dict,
+    ):
+
+        self._tabular_predictions = tabular_predictions
+        self._zeroshot_context = zeroshot_context
+        self._zeroshot_context.subset_datasets(self._tabular_predictions.datasets)
+        self._zeroshot_context.subset_models(self._tabular_predictions.list_models_available(present_in_all=True))
+        self._df_metadata = zeroshot_context.df_metadata if zeroshot_context.df_metadata is not None else pd.read_csv(Paths.data_root / "metadata" / "task_metadata.csv")
+        self._tid_to_name = dict(self._df_metadata[['tid', 'name']].values)
+        self._tid_to_name = {k: v for k, v in self._tid_to_name.items() if k in self._tabular_predictions.datasets}
+        self._name_to_tid = {v: k for k, v in self._tid_to_name.items()}
+        self._ground_truth = ground_truth
+        assert all(x in self._tid_to_name for x in self._tabular_predictions.datasets)
+
```

Comment:
in my update I change how this works a bit so that EvaluationRepository does not perform any operations on init and is simply a wrapper. For now I am ok to merge this code as is, but I think a wrapper will be better in general rather than performing these in-place updates on the input objects.

Reply:
I agree that ideally, there will be update such as `self._zeroshot_context.subset_datasets`, I think extracting tid dict is fine as it is readonly and quite simple.